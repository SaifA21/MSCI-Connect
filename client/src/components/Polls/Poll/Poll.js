import * as React from 'react';
import {Typography, Card, CardActions, CardContent, Grid, AppBar, Box, Toolbar, Button, Select, MenuItem, FormControl, InputLabel, TextField, Radio, FormControlLabel, RadioGroup, FormLabel, FormHelperText, helperText, TableRow} from "@material-ui/core/";
import ThumbUpAltIcon from '@material-ui/icons/ThumbUpAlt';
import ThumbDownIcon from '@material-ui/icons/ThumbDown';
import IconButton from '@material-ui/core/IconButton';


const serverURL = '';

const  Poll = (props) => {
    const onclick = () =>{
      console.log('clicked')
    }

    const [vote, setVote] = React.useState('');


    return (
      <Card sx={{ minWidth: 275 }}>
        <CardContent>
          <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
            Description: {props.description}
          </Typography>
          <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
            Option 1: {props.option1}, with {props.votes1} votes so far
          </Typography>
          <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
            Option 2: {props.option2}, with {props.votes2} votes so far
          </Typography>
          <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
            Option 3: {props.option3}, with {props.votes3} votes so far
          </Typography>
          <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
            Option 4: {props.option4}, with {props.votes4} votes so far
          </Typography>
          

        </CardContent>
        <CardActions>
            <FormControl>
            <Select
            
            onChange={(event)=>{        
              setVote(event.target.value)
              console.log(event.target.value)
             
  
            }}
            displayEmpty
            
            inputProps={{ 'aria-label': 'Without label' }}
            >
            <MenuItem value="" disabled>
                Placeholder
            </MenuItem>
            <MenuItem value={1}>{props.option1}</MenuItem>
            <MenuItem value={2}>{props.option2}</MenuItem>
            <MenuItem value={3}>{props.option3}</MenuItem>
            <MenuItem value={4}>{props.option4}</MenuItem>

            </Select>
            <FormHelperText>Placeholder</FormHelperText>
            </FormControl>

          <ButtonSubmitVote pollID = {props.pollID} vote = {vote} />
        </CardActions>
      </Card>
    );
  }

  const ButtonSubmitVote= (props) => {
   
    
    const addPollVote = () => {
      console.log(props.pollID)
      callAddPollVote(props)
      .then(res => {
          var parsed = JSON.parse(res.express);
         
        }
      ).then(console.log(props.vote))
    }
  
    const callAddPollVote = async (props) => {
      
      const url = serverURL + "/api/addPollVote";
      console.log(props.vote)
      const response = await fetch(url, {method: "POST", headers: {
        "Content-Type": "application/json",
      },body: JSON.stringify({vote: props.vote, pollID : props.pollID})});
      const body = await response.json();
      if (response.status !== 200) throw Error(body.message);
      return body;
    }
  

  
    
    
    const handleSubmitVote = () => {
      addPollVote();
      console.log(props.vote);
    };
    
  return(
    <div> 
      
      <Button onClick={handleSubmitVote}>Submit</Button>
    </div>
  )
}
 

  
  export default (Poll);
  


  //<ButtonMailingList update ={update} />

  /*
const ButtonMailingList= (props) => {

  const addMailingList =  () => {
    callApiAddMailingList()
      .then(res => {
        var parsed = JSON.parse(res.express);
      })
  
  } 
  
  const callApiAddMailingList = async () => {

    const url = serverURL + "/api/addMailingList"
  
    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(props.update)
  
    });
    const body = await response.json();
    if (response.status != 200) throw Error(body.update);
    return body;
  }

  const handleMailingList = () => {
    addMailingList();
  };

  return(
    <div> 
      <Button onClick={handleMailingList}>Subscribe to Mailing List</Button>
    </div>
  )
}



*/