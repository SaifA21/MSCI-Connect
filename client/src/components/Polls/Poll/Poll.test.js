import renderer from 'react-test-renderer';
import Poll from './Poll.js'


//Unit test written by Sajeen Selvakamalan

describe('Poll Component', () => {

    it('renders correctly', () => {
        const tree = renderer
          .create(<Poll></Poll>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

})