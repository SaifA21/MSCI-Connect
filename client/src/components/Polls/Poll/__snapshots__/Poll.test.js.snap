// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll Component renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  sx={
    Object {
      "minWidth": 275,
    }
  }
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      Description: 
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      Option 1: 
      , with 
       votes so far
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      Option 2: 
      , with 
       votes so far
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      Option 3: 
      , with 
       votes so far
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      Option 4: 
      , with 
       votes so far
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <div
      className="MuiFormControl-root"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-label="Without label"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        className="MuiFormHelperText-root"
      >
        Placeholder
      </p>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
</div>
`;
